lib/log.js

```diff
@@ -0,0 +1,94 @@
+var path = require('path')
+var fs = require('fs')
+var Console = require('console')
+var mkdir = require('mkdirp')
+var wsock = require('websocket-stream')
+var http = require('http')
+var through = require('through2')
+
+var nodeLogger = null
+var browserLogger = null
+
+var PORT = 17509
+
+module.exports.Node = function () {
+  if (!nodeLogger) {
+    nodeLogger = setupNodeLogger()
+  }
+  return nodeLogger
+}
+
+module.exports.Browser = function () {
+  if (!browserLogger) {
+    browserLogger = setupBrowserLogger()
+  }
+  return browserLogger
+}
+
+function setupNodeLogger () {
+  var electron = require('electron')
+
+  // create log dir
+  var logDir = path.join(electron.app.getPath('userData'), 'data', 'logs')
+  mkdir.sync(logDir)
+
+  // get electron app data dir
+  var filename = (new Date()).toISOString() + '.txt'
+  var logFilename = path.join(logDir, filename)
+
+  // fs write stream to file
+  var dest = fs.createWriteStream(logFilename)
+
+  // through stream to multiplex to stdout
+  var multi = multiplex([process.stdout, dest])
+
+  // create console instance
+  var log = new Console.Console(multi, null).log
+
+  // start ws server
+  websocketServer(function (socket) {
+    socket.on('data', function (text) {
+      multi.write(text)
+    })
+  })
+
+  return log
+}
+
+function setupBrowserLogger () {
+  var socket = wsock('ws://localhost:'+PORT)
```

Comment:
Great question. I tried, and it indeed broke the browser process. For now, I've wrapped the socket write in a try/catch so that failure isn't quite so catastrophic. In practice, the node process always starts before the browser window is created, so this hopefully won't be an issue too often. If we notice it is, we can write a module like [reconnect-ws](https://github.com/jacobbubu/reconnect-ws) that'll do reconnects, and also some logic to buffer what failed to be written until the reconnect happens.

(I tried out `reconnect-ws`, but it has native deps that are failing to compile. There's also a non-merged PR for buffering unwritable data for future writing, but it doesn't merge cleanly anymore. Might be a bit of a project..)